### PR TITLE
Adds an exploding chameleon flag for traitors

### PR DESCRIPTION
**Pull request by IcyV**
This gives traitors a 7TC chameleon flag. It can be used once to be
disguised as any other flag on the station. It comes with an explosive
payload equivalent to a minibomb. The timer for this is started once
the flag is lit on fire. By default it looks like a standard NT flag
that has a separate name and description by default.

:cl: IcyV
add: Adds an exploding chameleon flag for traitors.
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
RIP Escape/North of Bridge because someone lit the flag on fire. Sounds like fun!

File: code/game/objects/items/flag.dm [hunk 1]
```diff
@@ -219,3 +219,47 @@
...
+	var/list/flag_types = typesof(/obj/item/flag) - list(src.type, /obj/item/flag)
+	var/list/flag = list()
+
+	for(var/flag_type in flag_types)
+		var/obj/item/flag/F = new flag_type
+		flag[capitalize(F.name)] = F
+
+	var/list/show_flag = list("EXIT" = null) + sortList(flag)
+
+	var/input_flag = input(user, "Choose a flag to disguise as.", "Choose a flag.") in show_flag
```
Comment (monster860) [inline code/game/objects/items/flag.dm]:
```
var/input_flag = input(user, "Choose a flag to disguise as.", "Choose a flag.") as text|null in sortList(flag)
```

File: code/game/objects/items/flag.dm [hunk 2]
```diff
@@ -219,3 +219,47 @@
...
+		if(chosen_flag)
+			name = chosen_flag.name
+			icon_state = chosen_flag.icon_state
+			used = 1
+			
+/obj/item/flag/chameleon/process()
+	if(lit && (time > 0))
+		time -= 2
+	if(lit && (time <= 0))
+		explosion(src.loc,1,2,4,4, flame_range = 4)
```
Comment (monster860) [inline code/game/objects/items/flag.dm]: ```explosion(loc,1,2,4,4, flame_range = 4)```

No src.

File: code/game/objects/items/flag.dm [hunk 3]
```diff
@@ -219,3 +219,47 @@
...
+		var/obj/item/flag/F = new flag_type
+		flag[capitalize(F.name)] = F
+
+	var/list/show_flag = list("EXIT" = null) + sortList(flag)
+
+	var/input_flag = input(user, "Choose a flag to disguise as.", "Choose a flag.") in show_flag
+
+	if(user && src in user.contents)
+
+		var/obj/item/flag/chosen_flag = flag[capitalize(input_flag)]
```
Comment (monster860) [inline code/game/objects/items/flag.dm]: `capitalize()` is unnecessary here.

File: code/game/objects/items/flag.dm [hunk 4]
```diff
@@ -219,3 +219,47 @@
...
+
+		if(chosen_flag)
+			name = chosen_flag.name
+			icon_state = chosen_flag.icon_state
+			used = 1
```
Comment (monster860) [inline code/game/objects/items/flag.dm]: You should set desc as well.

File: code/game/objects/items/flag.dm [hunk 5]
```diff
@@ -219,3 +219,47 @@
...
+	origin_tech = "materials=3;magnets=4;syndicate=4"
+	var/time = 0
+	var/used = 0
+	
+/obj/item/flag/chameleon/New()
+	time = rand(0,30)
+	
+/obj/item/flag/chameleon/attack_self(mob/user as mob)
+	if(used)
+		return
```
Comment (monster860) [inline code/game/objects/items/flag.dm]: I don't see much point in making these only disguizable once.

File: code/game/objects/items/flag.dm (hunk 4 above)
Comment (IcyV) [inline code/game/objects/items/flag.dm]: Totally spaced on that, thanks.

File: code/game/objects/items/flag.dm (hunk 5 above)
Comment (IcyV) [inline code/game/objects/items/flag.dm]: Makes people think about their choice of flag and cuts down a bit from people just spamming use on flags for meta.

File: code/game/objects/items/flag.dm (hunk 1 above)
Comment (IcyV) [inline code/game/objects/items/flag.dm]: That just wants to throw out a bunch of errors for whatever reason.

**Comment by Fox-McCloud** [comment]
Going to conflict horrendously with the fire system rework. Probably best to wait on this until that's done.

**Comment by IcyV** [comment]
Fine with me. I largely wanted it up for opinions and tweaking in the meantime.

**Comment by Fox-McCloud** [comment]
@IcyV 

The fire system rips out a lot of the stuff you're utilizing here (like the `lit` var), so it may be best to wait. You can resolve merge conflicts, of course..but you heavily use the current "on fire" system for this flag and that's completely gone in the updated fire system.

**Comment by IcyV** [comment]
Righto, I'll see how it goes here then.

**Comment by Fox-McCloud** [comment]
yeah, this definitely won't work. The only reason it compiles is because I'm a derp and forgot to remove a var in the fire system port.

File: code/game/objects/items/flag.dm (hunk 5 above)
Comment (Iamgoofball) [inline code/game/objects/items/flag.dm]: thats dumb reasoning

just do it properly with an action button chameleon effect and dont snowflake it

Comment (IcyV) [inline code/game/objects/items/flag.dm]: @Iamgoofball Got an example I can peek at? Can't say I really know much on the code-side when it comes to action buttons and such.

**Pull request closed without merging**